Add ogc-api-records service to docker composition

jahow opened the pull request:
Currently not working

Addresses #145

jahow committed:
add ogc-api-records service in base docker composition

jahow committed:
add rules in traefik

jahow commented:
Requires https://github.com/georchestra/datadir/pull/279 to be merged

fvanderbiest commented:
Will also require a new mapping in https://github.com/georchestra/datadir/blob/docker-master/security-proxy/targets-mapping.properties

jeanmi151 commented:
@jahow  I am getting a crash after ogc start : 
`docker-ogc-api-records-1                    | Caused by: org.postgresql.util.PSQLException: ERROR: relation "languages" does not exist`

jahow committed:
make dependency from geonetwork to kibana not rely on a healthcheck

kibana is not essential for GN to startup correctly

jahow commented:
* rebased to include latest changes in the datadir
* fixed an error with a missing healthcheck on Kibana.

The service starts up correctly provided the database was populated originally by GeoNetwork. I still get 404 on all routes from Nginx on the docker composition though... @jeanmi151

jeanmi151 committed:
updating config dir

jeanmi151 committed:
updating submodule to use ssh

jeanmi151 commented:
connected with https://github.com/georchestra/datadir/pull/281

fvanderbiest commented on .gitmodules:
```diff
@@ -1,8 +1,8 @@
 [submodule "resources/static"]
 	path = resources/static
-	url = https://github.com/georchestra/htdocs.git
+	url = git@github.com:georchestra/htdocs.git
```
We should keep https here, as some firewalls are really tough with git.

jeanmi151 replied:
https is okay for none auth clone but not for everything it does no works (auth, modifications, pushing ...)

fvanderbiest commented on docker-compose.yml:
```diff
@@ -300,3 +295,16 @@ services:
+      LANG: en_US.UTF-8
+      SERVER_SERVLET_CONTEXT_PATH: /ogc-api-records
+      SPRING_CONFIG_LOCATION: file:///etc/georchestra/geonetwork/microservices/ogc-api-records/config.yml
+      SPRING_PROFILES_ACTIVE: standalone
```
`JAVA_OPTS: -Dfile.encoding=UTF-8` ? see https://github.com/georchestra/ansible/commit/55f87bcf5986de2e5dc7ed34256691a1d1d3c65a

jeanmi151 committed:
update ogc var env

jeanmi151 committed:
reverting submodules changes

jahow commented:
still needs a rebase apparently? anyway, thanks for picking this up :)

fvanderbiest commented on docker-compose.yml:
```diff
@@ -300,3 +295,17 @@ services:
       - ./resources/kibana/kibana.yml:/usr/share/kibana/config/kibana.yml
 
+  ogc-api-records:
+    image: geonetwork/gn-cloud-ogc-api-records-service:4.2.2
```
ideally: should be "latest" on master, but wondering also if this should be a `geonetwork/gn-cloud-ogc-api-records-service` or a `georchestra/gn-cloud-ogc-api-records-service`, see https://github.com/georchestra/georchestra/issues/3931

jeanmi151 replied:
I can set latest, 
otherwise not idea of which image to use (geonetwork or georchestra)

jeanmi151 replied:
latest tag does not exist ..

jahow replied:
I think we tested it to work with this tag; seeing how much trouble we have making this all work together, staying on a floating tag sounds unfeasible to me

jeanmi151 replied:
not possible i guess : 
georchestra/gn-cloud-ogc-api-records-service:latest ---> really old (1 year ago)
geonetwork/gn-cloud-ogc-api-records-service:latest ---> do not exist

jahow replied:
(I meant 4.2.2)

jeanmi151 committed:
update tag to latest

jeanmi151 committed:
add dependencies to ogc-api-records service

jeanmi151 commented:
Ready to land, waiting for reviews
LGTM and tested/working
please if you merge squash the commits

jahow committed:
revert change to kibana
